data/mods/gen4/random-teams.ts

```diff
@@ -498,98 +516,90 @@ export class RandomGen4Teams extends RandomGen5Teams {
 
 		const movePool = (species.randomBattleMoves || Object.keys(this.dex.data.Learnsets[species.id].learnset!)).slice();
 		const rejectedPool: string[] = [];
-		const moves: ID[] = [];
+		const moves = new Set<string>();
 		let ability = '';
 		let item: string | undefined;
 		const evs = {hp: 85, atk: 85, def: 85, spa: 85, spd: 85, spe: 85};
-		let ivs: SparseStatsTable = {hp: 31, atk: 31, def: 31, spa: 31, spd: 31, spe: 31};
+		const ivs: SparseStatsTable = {hp: 31, atk: 31, def: 31, spa: 31, spd: 31, spe: 31};
+
+		const types = new Set<string>();
+		types.add(species.types[0]);
+		if (species.types[1]) types.add(species.types[1]);
+
+		const abilities = new Set<string>();
+		abilities.add(species.abilities[0]);
+		if (species.abilities[1]) abilities.add(species.abilities[1]);
+		if (species.abilities.H) abilities.add(species.abilities.H);
```

Comment:
```suggestion
		const abilities = new Set(Object.values(species.abilities));
```
same situation as gen 3